code/modules/bluespace_exploration/system_console.dm

```diff
@@ -0,0 +1,159 @@
+			"alignment" = "Unknown",
+			"threat" = "Unknown",
+			"research_value" = "Unknown",
+			"distance" = calculate_distance_to_stationary_port(S),
+		)
+		data["stars"] += list(formatted_star)
+	return data
+
+/obj/machinery/computer/system_map/ui_act(action, params)
+	var/obj/docking_port/mobile/linkedShuttle = SSshuttle.getShuttle(shuttle_id)
+	if(!linkedShuttle || linkedShuttle.mode != SHUTTLE_IDLE)
+		say("Jump already in progress.")
+		return
+	switch(action)
+		if("jump")
+			var/obj/machinery/bluespace_drive/bs_drive = null
+			if(linked_bluespace_drive)
+				bs_drive = linked_bluespace_drive.resolve()
+			//Locate attached ship.
+			var/datum/ship_datum/attached_ship = SSbluespace_exploration.tracked_ships[shuttle_id]
+			if(!attached_ship)
+				say("Console not linked to a ship, please rebuild this console on a bluespace capable shuttle.")
+				return
+			//Locate Star
+			var/star_name = params["system_name"]
+			if(!star_name)
+				return
+			//Check if the jump location is actually a standard one
+			if(star_name in standard_port_locations)
+				handle_jump_to_port(star_name)
+				return
+			var/star = attached_ship.star_systems[star_name]
+			if(!star)
+				return
+			//Check for jumping actions
+			if(attached_ship.bluespace)
+				if(!bs_drive || QDELETED(bs_drive))
+					say("Your drive is experiencing issues, or cannot be located. Please contact your ship's engineer.")
+					return
+				//Locate the BS drive and then trigger jump
+				say("Sending engagement request to bluespace drive...")
+				bs_drive.engage(star)
+			else
+				handle_space_jump(star)
+			attached_ship.recalculate_star_systems()
+
+//Do this a few frames after loading everything, since if it loads at the same time as the drive it can fail to be located
+/obj/machinery/computer/system_map/proc/locate_bluespace_drive()
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttle_id)
+	if(M)
+		for(var/obj/machinery/bluespace_drive/BSD as anything in GLOB.bluespace_drives)
+			if(get_turf(BSD) in M.return_turfs())
+				linked_bluespace_drive = WEAKREF(BSD)
+				return
+
+/obj/machinery/computer/system_map/proc/get_attached_shuttle()
+	var/turf/our_turf = get_turf(src)
+	for(var/shuttle_dock_id in SSbluespace_exploration.tracked_ships)
+		var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttle_dock_id)
+		if(!M)
+			continue
+		if(M.z != z)
+			continue
+		if(our_turf in M.return_turfs())
+			shuttle_id = M.id
+			break
+
+/obj/machinery/computer/system_map/proc/handle_jump_to_port(static_port_id)
+	return
+
+/obj/machinery/computer/system_map/proc/handle_space_jump(star)
+	say("Calculating hyperlane, please stand back from the doors...")
+	SSbluespace_exploration.request_ship_transit_to(shuttle_id, star)
+
+/obj/machinery/computer/system_map/proc/calculate_distance_to_stationary_port(obj/docking_port/stationary/S)
+	var/deltaX = S.x - x
+	var/deltaY = S.y - y
+	var/deltaZ = (S.z - z) * 500
+	return sqrt(deltaX * deltaX + deltaY * deltaY + deltaZ * deltaZ)
```

Comment:
I assume this fine, brain too fried for math.

Reply:
Pythagoras' theorem